ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/ChainDB/Impl/ChainSel.hs

```diff
@@ -557,7 +557,7 @@ chainSelectionForBlock cdb@CDB{..} blockCache hdr punish = electric $ do
         traceWith addBlockTracer $ IgnoreBlockOlderThanK p
         return tipPoint
 
-      -- We might have validated the block in the meantime
+      -- The block is invalid
```

Comment:
I think ChainSel could host a long note explaining how LoE and GDD work, to which we can refer from the various places that talk about LoE and GDD.